prefect_aws/secrets_manager.py

```diff
@@ -350,3 +352,152 @@ def example_delete_secret_with_recovery_window():
         logger.exception("Unable to delete secret %s", secret_name)
         raise
+
+
+class SecretsManager(SecretBlock):
+    """
+    Block that represents a resource that can store and retrieve secrets.
+
+    Attributes:
+        aws_credentials: The credentials to use for authentication with AWS.
+        secret_name: The name of the secret.
+    """
+
+    aws_credentials: Union[AwsCredentials, MinIOCredentials]
+    secret_name: str = Field(default=..., description="The name of the secret.")
+
+    @sync_compatible
+    async def read_secret(
+        self,
+        version_id: str = None,
+        version_stage: str = None,
+        **read_kwargs: Dict[str, Any],
+    ) -> bytes:
+        """
+        Reads the secret from the secret storage service.
+
+        Args:
+            version_id: The version of the secret to read. If not provided, the latest
+                version will be read.
+            version_stage: The version stage of the secret to read. If not provided,
+                the latest version will be read.
+            read_kwargs: Additional keyword arguments to pass to the
+                `get_secret_value` method of the boto3 client.
+
+        Returns:
+            The secret data.
+
+        Examples:
+            Reads a secret.
+            ```python
+            secrets_manager = SecretsManager.load("MY_BLOCK")
+            secrets_manager.read_secret()
+            ```
+        """
+        client = self.aws_credentials.get_secrets_manager_client()
+        if version_id is not None:
+            read_kwargs["VersionId"] = version_id
+        if version_stage is not None:
+            read_kwargs["VersionStage"] = version_stage
+        response = await run_sync_in_worker_thread(
+            client.get_secret_value, SecretId=self.secret_name, **read_kwargs
+        )
+        secret = response["SecretBinary"]
+        arn = response["ARN"]
+        self.logger.info(f"The secret {arn!r} data was successfully read.")
+        return secret
+
+    @sync_compatible
+    async def write_secret(
+        self, secret_data: bytes, **put_or_create_secret_kwargs: Dict[str, Any]
+    ) -> str:
+        """
+        Writes the secret to the secret storage service as a SecretBinary;
+        if it doesn't exist, it will be created.
+
+        Args:
+            secret_data: The secret data to write.
+            **put_or_create_secret_kwargs: Additional keyword arguments to pass to
+                put_secret_value or create_secret method of the boto3 client.
+
+        Returns:
+            The path that the secret was written to.
+
+        Examples:
+            Write some secret data.
+            ```python
+            secrets_manager = SecretsManager.load("MY_BLOCK")
+            secrets_manager.write_secret(b"my_secret_data")
+            ```
+        """
+        client = self.aws_credentials.get_secrets_manager_client()
+        try:
+            response = await run_sync_in_worker_thread(
```

Comment:
Does `run_sync_in_worker_thread` work outside of a flow?

Reply:
Yes it works:
```
    call = partial(__fn, *args, **kwargs)
    return await anyio.to_thread.run_sync(
        call, cancellable=True, limiter=get_thread_limiter()
    )
```

Reply:
```
import anyio
def test():
    print("Hello")
await anyio.to_thread.run_sync(test, cancellable=True)
```